daphne/http_protocol.py

```diff
@@ -250,8 +250,9 @@ def handle_reply(self, message):
             # Write headers
             for header, value in message.get("headers", {}):
                 self.responseHeaders.addRawHeader(header, value)
-            if self.server.server_name and self.server.server_name.lower() != "daphne":
-                self.setHeader(b"server", self.server.server_name.encode("utf-8"))
+            
+            if self.server.server_name and not self.responseHeaders.hasHeader("server"):
+                self.setHeader(b"server", self.server.server_name.lower().encode("utf-8"))
```

Comment:
I don't see why we would use `.lower()` here.

Also we can rely on the default encoding being utf-8 on Python 3.

```suggestion
                self.setHeader(b"server", self.server.server_name.encode())
```